scripts/lib/CIME/case_st_archive.py

```diff
@@ -90,6 +88,9 @@ def _get_component_archive_entries(case, archive):
         if archive_entry is not None:
             yield(archive_entry, compname, archive_entry.get("compclass"))
 
+    if compname not in compset_comps:
+        logger.debug('Skipping compname %s; it is not in compset_comps' %(compname))
```

Comment:
My earlier comment still stands here:

> I don't understand the intent of this. Will this ever get triggered? I'm confused because, in the above loop, compname takes on the various values of compset_comps. I can't understand any scenario where compname down here would not be something in compset_comps. Maybe this was a debugging statement that you meant to remove?

Can you either remove these two lines or explain the intent?

Reply:
I did put it in as a debugging statement because there were problems
implementing short-term archiving of component esp before it was 'esp'.
There are other debugging statements in this module like this one,
so I thought it would be useful to leave it in.  If it doesn't fit with the philosophy
of the scripts, I'm happy to take it out.
Kevin

Reply:
I don't mind debugging statements in general - I'm just having a hard time seeing how this one could possibly get triggered: compname just loops through compset_comps, so under what circumstances could it ever be the case that compname is not in compset_comps?

Reply:
Doh!  I moved this from another, superseded routine, where compname was defined
independently of the for loop.  I don't know where it would fit now, so I'll delete it.
Thanks for being persistent.